src/preload.py

```diff
@@ -610,7 +610,17 @@ def write_resin_device_pinning(app_data, output):
 
 
 def write_apps_json(data, output):
-    """Writes data dict to output as json"""
+    """Updates or creates preloaded state"""
+    if os.path.isfile(output):
+        with open(output, "r") as f:
+            current_state = json.load(f)
+        for uuid in data.keys():
+            if len(uuid) != len(current_state.keys()[0]):
```

Comment:
I understand that `current_state` is the existing `apps.json` in the image file, and `data` is data loaded from the API. So in the expression `for uuid in data.keys()`, currently/today, `uuid` is actually a release ID, right? The release ID being a number like 12345678, rather than a UUID.

To confirm my understanding:
* Future PRs to `balena-preload` (or `balena-cli` or both) will be required for `uuid` to actually be a UUID (as it is currently a release ID).
* Before this PR, the CLI could not be used with new hostapp-aware images because `apps.json` would be overwritten (not because of ID vs UUID, which is a supervisor matter).
* After this PR, the CLI still can be used with new hostapp-aware images because `apps.json` will be merged, even if release IDs (rather than UUIDs) are still used.

Reply:
Old images that are not hostapp aware don't have any existing `apps.json` file, or do they?
In which case, `current_state` would not be defined and `current_state.keys()` would raise an exception?

Reply:
@pdcastro in `for uuid in data.keys()`, data is of the form:
```
{
    "apps": {
        "119711": {
            "commit": "d99be5beb5cffd74af068cb4f4f55389",
            "name": "rpi4_idle",
            "networks": {},
            "releaseId": 133467,
```
So the `uuid` is currently the appID (119711), not the releaseID (133467).

From your points:

* Future PRs to balena-preload (or balena-cli or both) will be required for uuid to actually be a UUID (as it is currently an app ID). This change is also happening in the UI for a v3 target state.
* Before this PR, the CLI could not be used with new hostapp-aware images because apps.json would be overwritten, and also because the SV required in those images require apps.json to use appUUID and not appID.
After this PR, the CLI still cannot be used with new hostapp-aware images because the preloaded app does not use appUUIDs.

I will leave this as a draft until we have the whole picture.